Linux ppc64le: libffi FFI_DEFAULT_ABI mismatch — ffi_prep_cif always returns FFI_BAD_ABI

### Version

3.4.1

### Platform

Linux ppc64le

### JDK

Red Hat OpenJDK 25.0.3 (ppc64le) — but JDK-independent

### Module

LWJGL core (libffi)

### Bug description

## Bug description

On **Linux ppc64le**, the value of `LibFFI.FFI_DEFAULT_ABI` baked into the bindings is
**wrong**: it is computed from libffi's 32-bit *SYSV* ABI enum instead of the 64-bit
*LINUX* (POWERPC64) enum. As a result it does not match the ABI enum that the bundled
`libffi.a` was actually built with, and **every** `ffi_prep_cif()` / callback creation
fails with `FFI_BAD_ABI` (2).

This makes the official ppc64le natives effectively unusable for anything that creates a
native callback. For example, Minecraft crashes on startup the moment it installs its
GLFW error callback:

```
Caused by: java.lang.IllegalStateException: Failed to prepare libffi CIF: 2
    at org.lwjgl.system.APIUtil.apiCreateCIF(APIUtil.java:633)
    at org.lwjgl.glfw.GLFWErrorCallbackI.<clinit>(GLFWErrorCallbackI.java:23)
```

…but the bug is in LWJGL core, not GLFW or Minecraft — see the minimal repro below,
which uses **only `lwjgl` core**.

### The two values disagree

LWJGL exposes the default ABI two ways:

* `LibFFI.FFI_DEFAULT_ABI` — a constant baked when compiling the binding wrapper
  `org_lwjgl_system_libffi_LibFFI.c` (`return (jint)FFI_DEFAULT_ABI;`).
* `LibFFI.ffi_get_default_abi()` — a function implemented **inside libffi itself**
  (`ffi.h`: `FFI_API unsigned int ffi_get_default_abi(void)`), i.e. compiled into
  `libffi.a`.

Your own test `org.lwjgl.system.libffi.LibFFITest#testConstants` asserts these are equal:

```java
assertEquals(ffi_get_default_abi(), FFI_DEFAULT_ABI);
```

**On ppc64le this assertion fails.** It evidently never runs there — CI cross-compiles
ppc64le but does not execute the test suite on ppc64le hardware/emulation.

### Minimal, complete, verifiable example

`PpcFfiBug.java` — depends only on `lwjgl` core + its ppc64le natives:

```java
import org.lwjgl.system.libffi.*;
import static org.lwjgl.system.libffi.LibFFI.*;

public class PpcFfiBug {
    public static void main(String[] args) {
        System.out.println("os.arch                                 = " + System.getProperty("os.arch"));
        System.out.println("FFI_DEFAULT_ABI (binding constant)      = " + FFI_DEFAULT_ABI);
        System.out.println("ffi_get_default_abi() (inside libffi.a) = " + ffi_get_default_abi());
        System.out.println("FFI_LAST_ABI    (binding constant)      = " + FFI_LAST_ABI);

        FFICIF cif = FFICIF.malloc();
        int status = ffi_prep_cif(cif, FFI_DEFAULT_ABI, ffi_type_void, null);
        System.out.println("ffi_prep_cif(FFI_DEFAULT_ABI) = " + status +
            (status == FFI_OK ? " (FFI_OK)" : " (FFI_BAD_ABI -> every callback/CIF fails)"));
        cif.free();
    }
}
```

Run it (downloading the official 3.4.1 ppc64le natives):

```bash
BASE=https://build.lwjgl.org/release/3.4.1/bin
curl -sLO https://repo1.maven.org/maven2/org/lwjgl/lwjgl/3.4.1/lwjgl-3.4.1.jar
curl -sLo lwjgl-natives.jar $BASE/lwjgl/lwjgl-natives-linux-ppc64le.jar

javac -cp lwjgl-3.4.1.jar PpcFfiBug.java
java  -cp .:lwjgl-3.4.1.jar:lwjgl-natives.jar PpcFfiBug
```

**Actual output (LWJGL 3.4.1, Linux ppc64le):**

```
os.arch                                 = ppc64le
FFI_DEFAULT_ABI (binding constant)      = 28
ffi_get_default_abi() (inside libffi.a) = 11
FFI_LAST_ABI    (binding constant)      = 32
ffi_prep_cif(FFI_DEFAULT_ABI) = 2 (FFI_BAD_ABI -> every callback/CIF fails)
```

**Expected:** `FFI_DEFAULT_ABI == ffi_get_default_abi()` and `ffi_prep_cif(...) == FFI_OK`.

This is reproducible by anyone on any ppc64le environment (real hardware or
`qemu-user` ppc64le) — it's a property of the published artifact, not of a particular
distro or `long double` configuration.

### Root cause

`modules/lwjgl/core/src/main/c/libffi/ppc64le/ffitarget.h` selects the 64-bit ABI enum
only when `POWERPC64` is defined:

```c
#if defined (POWERPC) && defined (__powerpc64__)   /* linux64 */
#ifndef POWERPC64
#define POWERPC64
#endif
...
#endif
```

`__powerpc64__` is a compiler builtin, but **`POWERPC` is not** — it is normally provided
by libffi's own autotools build (`-DPOWERPC`). When the LWJGL build compiles the binding
wrapper, it does **not** pass `-DPOWERPC`, so `POWERPC64` is never defined and the header
falls into the **32-bit SYSV** branch:

* SYSV branch → `FFI_DEFAULT_ABI` from `{FFI_SYSV=8, …, FFI_SYSV_LONG_DOUBLE_128=16}`,
  `FFI_LAST_ABI = 32` → yields **28**.
* `libffi.a` is built by autotools *with* `POWERPC` → 64-bit LINUX branch
  (`FFI_LINUX=8`, …, `FFI_LAST_ABI=16`) → its `ffi_get_default_abi()` returns **11**
  and `ffi_prep_cif` rejects anything `>= 16` (so 28 → `FFI_BAD_ABI`).

So the binding wrapper and the linked `libffi.a` are compiled from **two different
branches of the same header**.

`config/linux/build.xml` already does the analogous thing for x86:

```xml
<arg value="-DX86_64" if:set="build.arch.x64"/> <!-- for libffi/x86/ffitarget.h -->
```

ppc64le simply never got its equivalent.

### Suggested fix

Add `-DPOWERPC` for ppc64le in `config/linux/build.xml`, alongside the x86 case:

```xml
<arg value="-DX86_64"  if:set="build.arch.x64"/>      <!-- for libffi/x86/ffitarget.h -->
<arg value="-DPOWERPC" if:set="build.arch.ppc64le"/>  <!-- for libffi/ppc64le/ffitarget.h: selects the 64-bit ABI branch -->
```

With this, the wrapper and `libffi.a` are compiled from the same 64-bit branch, the two
ABI values agree, `LibFFITest#testConstants` passes, and `ffi_prep_cif` succeeds.

I'm happy to open a PR with this one-line change.

### Verification of the fix without rebuilding

`FFI_DEFAULT_ABI()` in the shipped `liblwjgl.so` is a single instruction
(`li r3,28; blr`). Patching the immediate `28 → 11` (or `15`) — a value the bundled
`libffi.a` accepts — makes `ffi_prep_cif` return `FFI_OK` and lets the game run, which
confirms the diagnosis end to end.



### Stacktrace or crash log output

```shell
Backend library: LWJGL version 3.4.1+2
Operating System: Linux (ppc64le) version 7.0.11-200.fc44.ppc64le
Java Version: 25.0.3, Red Hat, Inc.

java.lang.ExceptionInInitializerError
    at com.mojang.blaze3d.platform.GLX._initGlfw(GLX.java:46)
    at com.mojang.blaze3d.systems.RenderSystem.initBackendSystem(RenderSystem.java:179)
    at net.minecraft.client.Minecraft.<init>(Minecraft.java:503)
    at net.minecraft.client.main.Main.main(Main.java:232)
Caused by: java.lang.IllegalStateException: Failed to prepare libffi CIF: 2
    at org.lwjgl.system.APIUtil.apiCreateCIF(APIUtil.java:633)
    at org.lwjgl.system.APIUtil.apiCreateCIF(APIUtil.java:618)
    at org.lwjgl.glfw.GLFWErrorCallbackI.<clinit>(GLFWErrorCallbackI.java:23)
    ... 7 more
```

--- a/config/linux/build.xml
+++ b/config/linux/build.xml
@@ -186,6 +186,7 @@
                 <arg value="-I${src.main.rel}/libffi/riscv64" if:set="build.arch.riscv64"/>
                 <arg value="-I${src.main.rel}/libffi/x86" if:set="build.arch.x64|x86"/>
                 <arg value="-DX86_64" if:set="build.arch.x64"/> <!-- for libffi/x86/ffitarget.h -->
+                <arg value="-DPOWERPC" if:set="build.arch.ppc64le"/> <!-- for libffi/ppc64le/ffitarget.h -->
                 <arg value="-I${src.main.rel}/${platform}/liburing"/>
                 <arg value="-I${src.main.rel}/${platform}/liburing/include"/>
                 <fileset dir=".">

